### PR TITLE
HZ-433 - TO_EPOCH_MILLIS SQL function

### PR DESCRIPTION
**Pull request by ivanthescientist**
Adds support for `TO_EPOCH_MILLIS` function, which takes arbitrary `TIMESTAMP WITH TIME ZONE` and returns its Epoch Millisecond value in a `BIGINT`. 
Reverse conversion function to complement `TO_TIMESTAMP_TZ` function introduced in #18856.

Example query: 
```
SELECT TO_EPOCH_MILLIS(TO_TIMESTAMP_TZ(v)) FROM TABLE (generate_series(1,3))
```

Example Result: 
```
1000
2000
3000
```
Note that by default `TO_TIMESTAMP_TZ` interprets values as in seconds.

### CODE REVIEW

**Comment by viliam-durina** [comment]
run-lab-run

File: hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/NestingAndCasingExpressionTest.java [hunk 1]
```diff
@@ -469,7 +469,11 @@ public void test_COALESCE() {
 
     @Test
     public void test_TO_TIMESTAMP_TZ() {
-        check(sql("TO_TIMESTAMP_TZ(?) || TO_TIMESTAMP_TZ(?)"), 1L, 1L);
+        check(sql("TO_TIMESTAMP_TZ(?) || TO_TIMESTAMP_TZ(1)"), 1L);
```
Comment (viliam-durina) [inline hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/NestingAndCasingExpressionTest.java:472]:
```suggestion
        check(sql("TO_TIMESTAMP_TZ(?) || TO_TIMESTAMP_TZ(?)"), 1L, 1L);
```

**Review by viliam-durina — APPROVED** [review]

**Comment by hakanmemisoglu** [comment]
@ivanthescientist The tests which use `SYSTEM_OFFSET_MILLIS` are failing while running at my machine.

**Comment by ivanthescientist** [comment]
Nice catch @hakanmemisoglu! This was a bug with the way I was getting the system offset, it didn't account for changes to DST rules over time (some of the timezones stopped using DST not so long ago actually, I was getting offset for current date, while offset might have been different just few years ago due to active DST rules).

**Review by hakanmemisoglu — APPROVED** [review]

**Pull request merged**